### PR TITLE
feature/add find gender controller

### PR DESCRIPTION
**Pull request by borovHokage**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/966?issue_count=22&issue_position=3&next_issue_id=964&prev_issue_id=967

### CODE REVIEW

File: src/UserService.Business/Commands/Gender/FindGenderCommand.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      IGenderRepository genderRepository,
+      IGenderInfoMapper mapper)
+    {
+      _genderRepository = genderRepository;
+      _mapper = mapper;
+    }
+    public async Task<FindResultResponse<GenderInfo>> ExecuteAsync(FindGendersFilter filter)
+    {
+      FindResultResponse<GenderInfo> response = new();
+      response.Body = new();
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Gender/FindGenderCommand.cs:30]: do you need it?

File: src/UserService.Business/Commands/Gender/FindGenderCommand.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+      (List<DbGender> dbGenders, int totalCount) = await _genderRepository.FindGendersAsync(filter);
+
+      response.TotalCount = totalCount;
+
+      response.Body = _mapper.Map(dbGenders);
+
+      response.Status = response.Errors.Any()
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Gender/FindGenderCommand.cs:38]: there will be no errors anyway

File: src/UserService.Business/Commands/Gender/FindGenderCommand.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    private readonly IGenderInfoMapper _mapper;
+
+    public FindGenderCommand(
+      IGenderRepository genderRepository,
+      IGenderInfoMapper mapper)
+    {
+      _genderRepository = genderRepository;
+      _mapper = mapper;
+    }
+    public async Task<FindResultResponse<GenderInfo>> ExecuteAsync(FindGendersFilter filter)
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Gender/FindGenderCommand.cs:27]: you get exception if takeCount <= 0, add validation

File: src/UserService/Controllers/GenderController.cs [hunk 4]
```diff
@@ -18,5 +22,13 @@ public class GenderController : ControllerBase
...
       return await command.ExecuteAsync(request);
     }
+
+    [HttpGet("find")]
+    public async Task<FindResultResponse<GenderInfo>> FindAsync(
+      [FromServices] IFindGenderCommand command,
+      [FromQuery] FindGendersFilter request)
+    {
+      return await command.ExecuteAsync(request);
+    }
```
Comment (MariyaSin) [inline src/UserService/Controllers/GenderController.cs:32]: update spec version in startup

File: src/UserService.Business/Commands/Gender/FindGenderCommand.cs (hunk 1 above)
Comment (borovHokage) [reply src/UserService.Business/Commands/Gender/FindGenderCommand.cs]: removed

File: src/UserService.Business/Commands/Gender/FindGenderCommand.cs (hunk 2 above)
Comment (borovHokage) [reply src/UserService.Business/Commands/Gender/FindGenderCommand.cs:51]: fixed

File: src/UserService.Business/Commands/Gender/FindGenderCommand.cs (hunk 3 above)
Comment (borovHokage) [reply src/UserService.Business/Commands/Gender/FindGenderCommand.cs:36]: fixed

File: src/UserService/Controllers/GenderController.cs (hunk 4 above)
Comment (borovHokage) [reply src/UserService/Controllers/GenderController.cs:32]: done

File: src/UserService.Business/Commands/Gender/FindGenderCommand.cs (hunk 2 above)
Comment (MariyaSin) [reply src/UserService.Business/Commands/Gender/FindGenderCommand.cs:51]: ???

**Review by MariyaSin — APPROVED** [review]

**Review by asyagraf — APPROVED** [review]

**Review by grinakate — APPROVED** [review]

**Review by VladSpitsin — APPROVED** [review]

**Review by actimellka — APPROVED** [review]

**Pull request merged**